client/src/app/pages/main-layout/main-layout.component.html

```diff
@@ -1,41 +1,25 @@
-@if (!isLoggedIn()) {
-  <div class="top-1 bg-gray-700 text-white rounded-b-2xl z-50 text-center flex items-center justify-center gap-1">
-    You are currently using Helios in readonly mode.
-    <button (click)="login()" class="p-0 text-white hover:text-gray-300 underline">Login here.</button>
-  </div>
-}
-
 <main class="h-screen overflow-hidden">
-  <div class="flex h-full p-3">
+  @if (!isLoggedIn()) {
+    <!-- Top Bar -->
+    <div class="top-1 bg-gray-700 text-white rounded-b-2xl z-50 text-center flex items-center justify-center gap-1">
+      You are currently using Helios in readonly mode.
+      <button (click)="login()" class="p-0 text-white hover:text-gray-300 underline">Login here.</button>
+    </div>
+  }
+  <div class="flex flex-grow p-3" [ngClass]="isLoggedIn() ? 'h-[calc(100vh-24px)]' : 'h-[calc(100vh-48px)]'">
```

Comment:
best-practice: please move the logic to a computed function in the respective ts file